src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/glide/MauiCustomViewTarget.java

```diff
@@ -5,39 +5,99 @@
-public class MauiCustomViewTarget extends CustomViewTarget<ImageView, Drawable> {
     private final ImageLoaderCallback callback;
+    private final String resourceLogIdentifier;
     private boolean completed;
 
-    public MauiCustomViewTarget(@NonNull ImageView view, ImageLoaderCallback callback) {
+    public MauiCustomViewTarget(@NonNull ImageView view, ImageLoaderCallback callback, Object model) {
         super(view);
 
         this.completed = false;
         this.callback = callback;
+
+        if (logger.isVerboseLoggable && model != null) {
+            this.resourceLogIdentifier = model.toString();
+        } else {
+            this.resourceLogIdentifier = null;
+        }
+    }
+
+    public void load(RequestBuilder<Drawable> builder) {
+        // Glide does not support starting new loads from Target callbacks and throws an exception.
+        // https://github.com/bumptech/glide/blob/9c9f56f29e63114c409184dd1e82b42b3f0063f7/library/src/main/java/com/bumptech/glide/request/SingleRequest.java#L301-L311
+        //
+        // > You can't start or clear loads in RequestListener or Target callbacks.
+        // > If you're trying to start a fallback request when a load fails, use RequestBuilder#error(RequestBuilder).
+        // > Otherwise consider posting your into() or clear() calls to the main thread using a Handler instead.
+        //
+        // This only happens when the target can load the previous request, and that is only possible within a `CustomViewTarget`
+        // where the Request is being stored inside the target's view as a Tag.
+        // https://github.com/bumptech/glide/blob/9c9f56f29e63114c409184dd1e82b42b3f0063f7/library/src/main/java/com/bumptech/glide/request/target/CustomViewTarget.java#L226-L239
+        //
+        // For this reason, if someone awaits the result of the image loading and then starts a new request,
+        // we must ensure the new load happens in a new thread loop.
+        //
+        // Why don't we simply `post` all the callback invocations?
+        // - performance: 99% of the times we simply map the image source once
+        // - consistency: the callback assumes the load succeeded so it must run right away in the same loop
+        //                because there's a change someone messes up with the drawable in the mean time
+        //
+        // Why don't we `post` `setImageDrawable + callback` together?
+        // - performance: 99% of the times we simply map the image source once
+        // - consistency: Glide is race-condition free but only if we set the image directly in the `onResourceReady`
+        //
+        // This use case is covered by `LoadDrawableAsyncReturnsWithSameImageAndDoesNotHang` core device test.
+        if (getIsInvokingCallbacks()) {
+            if (logger.isVerboseLoggable) logger.v("defer load: " + resourceLogIdentifier);
+            this.view.post(() -> load(builder));
+            return;
+        }
+
+        builder.into(this);
     }
```

Comment:
@jonathanpeppers @PureWeen it took me a while but I finally fully understand what's happening and given the effort I thought it was good to add a huge comment to explain the situation to future developers :D